tests/contrib/aiobotocore/utils.py

```diff
@@ -30,8 +32,18 @@ def aiobotocore_client(service, tracer):
         aws_secret_access_key="aws",
         aws_session_token="aws",
     )
-    Pin.override(client, tracer=tracer)
-    try:
-        yield client
-    finally:
-        client.close()
+
+    # In aiobotocore 1.x, ClientCreatorContext was added: https://github.com/aio-libs/aiobotocore/pull/659
+    if hasattr(aiobotocore.session, "ClientCreatorContext") and isinstance(
```

Comment:
I'm not sure I understand the logic here, it seems like we do basically the same thing regardless of `if hasattr(aiobotocore.session, "ClientCreatorContext") and isinstance(client, aiobotocore.session.ClientCreatorContext )` . Could you explain it to me @wantsui ?